[daw-json-link] New port

mheyman opened the pull request:
**Describe the pull request**

- #### What does your PR fix?  
  New port for **json-link** (perhaps the fastest JSON deserializer/serializer). Because we don't yet have enough JSON libraries...

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all (I hope - only checked windows and linux). No need to update ci.baseline.txt.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
 Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes.

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

Commit:
[json-link] package exists

Neumann-A committed:
[vcpkg-tools] update nuget to 5.9.1 (#18447)

NancyLi1013 committed:
[vcpkg_download_distfile] Add condition for padding the SHA (#18459)

traversaro committed:
[open6254] Enable uwp support (#18390)

* open62541: Enable uwp support

* Update versions for open62541

NancyLi1013 committed:
[aubio] Add ws2_32 to linkage (#18385)

* [aubio] Add ws2_32 to linkage

* Update version files

PhoebeHui committed:
[devil] Fix ilut header (#18384)

* [devil Fix ilut header

* Update baseline

Neumann-A committed:
[libpq] Add Secur32.lib and openssl to libpq cmake wrapper (#18377)

* [libpq] add secur32.lib to wrapper

* version stuff

* add openssl fix.

* fix version stuff

MichaelGoulding committed:
[ryml] [c4core] Update RapidYaml (ryml) to newer version (#18230)

* remove old port version

* fix versions yet again

Co-authored-by: Michael Goulding <Michael.Goulding@microsoft.com>

myd7349 committed:
[yyjson] Update to v0.3.0 (#18359)

* [yyjson] Update to 0.3.0

* [yyjson] vcpkg x-add-version yyjson

Pospelove committed:
[raylib] Fix MinGW build (#18362)

.

ianmcook committed:
[arrow] Update to 4.0.0 (#17975)

* Update arrow to 4.0.0

* Format

* Try fix thrift

* Update versions/ files

* Do not set ZSTD_ROOT

* Remove double quotes causing Windows problems

* Apply patches

* Remove LIB_DIR_OPTIONS

* Tweak zstd flags

* Update version hash

* Format

* Fail early on x86

* Update hash

* Fail early on arm, arm64

* Update hash

* Add expected failures to to scripts/ci.baseline.txt

* Exclude mallocs from default features

* Update hash

* Set default-features to false for aws-sdk-cpp

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

* Specify only x64 support in manifest

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

* Remove unneeded ci.baseline.txt entries

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

* Remove dataset from default-features

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

* Update hash

* Remove zstd path args

* Update hash

Co-authored-by: Tanguy Fautre <tanguy@fautre.com>
Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

JonLiu1993 commented on ports/json-link/portfile.cmake:
```diff
@@ -0,0 +1,29 @@
+
+file(REMOVE_RECURSE
+    ${CURRENT_PACKAGES_DIR}/debug
+    ${CURRENT_PACKAGES_DIR}/lib
+    ${CURRENT_PACKAGES_DIR}/share/json_link
```
@mheyman ，Could you clarify why these two directories were deleted

mheyman replied:
The contents of those directories seem to land in `share/json-link` and those directories are left empty at the end of the port install. Note: something is still amiss because I cannot seem to use the *...config.cmake* packages placed in `json-link` but, as this is a header-only library, those package definitions are not needed in projects that use json-link.

I see the windows build on github is failing but haven't had an opportunity to dig into it (it succeeds at home).

JackBoosY committed:
[minizip] Fix usage, fix minizipConfig.cmake (#18472)

* [minizip] Fix usage, fix minizipConfig.cmake

* update version record

JonLiu1993 commented:
@mheyman ,Could you please take a look?
[config-x86-windows-out.log](https://github.com/microsoft/vcpkg/files/6667099/config-x86-windows-out.log)

dg0yt committed:
[windres-rc] Fix wrapper output (#17045)

Sibras committed:
[ffmpeg] Warn/fail when selecting unsupported features. (#17153)

* [ffmpeg] Warn/fail when selecting unsupported features.

* Update ffmpeg.json

* Bump port-version

* Update ffmpeg.json

* [ffmpeg] Check for gpl/nonfree aswell as all option.

* avisynth now support static linking

* fixup typo

* Update ffmpeg.json

* Revert to fatal error on even when all is selected

* Update ffmpeg.json

* [ffmpeg] Disable openh264 on uwp

* update

* Update versions/f-/ffmpeg.json

* update

* Update ffmpeg.json

* update

* update

* Update

Co-authored-by: Billy Robert ONeal III <bion@microsoft.com>
Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>

mheyman commented:
The failure looks like it occurs in a cmake `FetchContent_MakeAvailable()` call. This call is identical for linux and windows. The failure occurs after a `find_package(Git QUIET)` fails to find `git`. I suspect `git` is available in the path for linux and osx but not for windows builds.

Darrell Wright's *json-link* library pulls in headers from two other of his projects. That is what the `FetchContent_MakeAvailable()` is trying to do. 

If `FetchContent_MakeAvailable()` (with a requirement on git) is unavailable for vcpkg builds, it is possible that I can rework this port into three ports, one for each project. I'd have to patch-out the calls to `FetchContent...()`. If I do that, I think I should rename the port from *json-link* to *daw-json-link* with a dependency on *daw-utf-range* and *daw-header-libraries* to make it more obvious they are related (*daw-utf-range* is also dependent on *daw-header-libraries*). Much of the header files already land under a `daw/` directory (I'm a little dissapointed they don't all land under `daw/`).

Is that how I should proceed? Make three ports that never call `FetchContent_MakeAvailable()`? Or, is there an easier solution?

JonLiu1993 committed:
[uwebsockets] update to 19.2.0 (#18525)

* [uwebsockets] update to <v19.2.0>

* update version

mcmtroffaes committed:
[libass] fix fontconfig dependency in .pc file (#18516)

* [libass] fix fontconfig dependency in .pc file on x64-linux

* [libass] bump port version

* [libass] x-add-version

NancyLi1013 committed:
[libgpg-error] Remove COPYING.LIB from lib folder (#18492)

* [libgpg-error] Remove COPYING.LIB from lib folder

* Update version files

NancyLi1013 committed:
[flashlight-cuda] Fix installation (#18489)

* [flashlight-cuda] Fix installation

* Update version files

gjasny committed:
[function2] Update to version 4.2.0 (#18487)

Ryan-rsm-McKenzie committed:
[infoware] update to latest commit (#18486)

* update infoware to latest commit

* update deprecated functions

* remove redundant file cleanup

* add versioning

JackBoosY commented:
@mheyman  For dependent third-party libraries, we have two ways:
1. Use the port in vcpkg if it exists.
2. Add `vcpkg_from_github` or other download methods to pre-download in `portfile.cmake`, and copy the decompressed folder to source path.

Neargye committed:
[nameof] Update to 0.10.1 (#18574)

cenit committed:
[OpenCV] update to 4.5.2 (#18572)

* [OpenCV] update to 4.5.2

* update version refs

* [OpenCV4] fix qt integration

* fix refs

Commit:
[qt5-*] Make QT and dependents build on arm macos (#18567)

* [qt5-*] Make it build on ARM64 macos

* [qt5-base] update port version

c72578 committed:
[pcre2] Update to 10.37 (#18547)

- New upstream version 10.37
- Drop patches (fixed upstream):
    pcre2-10.35_add_check_for_Intel_CET.patch
    pcre2-10.35_fix_postfix_for_debug_Windows_builds.patch
    pcre2-10.35_fix-space.patch
- Switch from CONTROL to vcpkg.json
    .\vcpkg format-manifest .\ports\pcre2\CONTROL

NancyLi1013 committed:
[openblas] Fix the macro collision with Windows 10 SDK (#18530)

* [openblas] Fix the macro collision with Windows 10 SDK

* Update version files

* Update feature thread as threads in vcpkg.json

* Update versions/o-/openblas.json

daschuer committed:
[pthreads] Fix debug dll look-up path (#18528)

SvenPStarFinanz committed:
[gtest] Update to 1.11.0 (#18509)

* [gtest] Update to release version 1.11.0

* x-add-version

* Update ports/gtest/vcpkg.json

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

* renewal of x-add-version.

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

m-kuhn committed:
[librttopo] Allow building for Android  (#18451)

* [librttopo] Allow building for android

* x-add-version

* Update versions/l-/librttopo.json

Co-authored-by: Phoebe <20694052+PhoebeHui@users.noreply.github.com>

Co-authored-by: Phoebe <20694052+PhoebeHui@users.noreply.github.com>

Neumann-A committed:
[Qt6] Update to 6.1.1 (#18320)

* [qt6] Update to 6.1.1

* version stuff

* fix empty dir

* update version

* fix casing in qtbase

* [libpq] add secure32.lib on windows

* add clang-cl fix

* version stuff

* fix flag removal by adding a space

* version stuff

* make qt6->Qt6 usage consistent since Qt6 refers to itself as Qt6 with uppercase Q

* fix libpq wrapper opensll linkage on linux

* use policy to fix dependent ports

* version stuff

* use upstream patch

* version stuff

* fix source location for clang-cl

* version stuff

* remove ws change

* version stuff

m-kuhn committed:
[libspatialite] Fix building for android (#18452)

* [libspatialite] Fix building for android

* x-add-version

* Update versions/l-/libspatialite.json

* Update libspatialite.json

Commit:
[seal]Updated to version 3.6.6. Replaced CONTROL with manifest file. (#18457)

* Updated seal to version 3.6.6. Replaced CONTROL with manifest file.

* Changed version-string to version-semver.

* Updated versions/s-/seal.json.

* Added a missing CMake arguement.

* Updated versioning.

* Fixed hexl config file.

* Updated versioning.

* Fixed hexl not finding dependency.

* Updated versioning.

owent committed:
[libcopp] Update to 1.4.1 (#18425)

walbourn committed:
[dxut, fx11, directxmath] ports updated (#18431)

* [dxut] port updated for latest servicing

* update baseline

* Update to use latest DirectXMath

* Fix baseline

* fx11 metadata update

* Updated baseline

* Converted directxmath CONTROL to vcpkg.json

* Update baseline

* Apply suggestions from code review

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

* Code review feedback

* Update baseline

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

walbourn committed:
[uvatlas, spectra] ports updated (#18404)

* [uvatlas, spectra] ports updated

* Update versions

* Minor code review

* Update version

* Code review feedback

* Update baseline

danielaparker committed:
[jsoncons] Update to 0.165.0 (#18284)

* [jsoncons] Update to 0.165.0

* [jsoncons] Update to 0.165.0, update baseline

* [jsoncons] Added vcpkg_fixup_pkgconfig() to portfolio.cmake

* update version

Co-authored-by: Jonliu1993 <13720414433@163.com>

NancyLi1013 committed:
[cpr] Update to 1.6.2 (#18386)

* [cpr] Update to 1.6.2

* Update version files

* Update patch

* Update verison files

* Update ports/cpr/portfile.cmake

* Update versions/c-/cpr.json

* Fix build error on uwp

* Update versions/c-/cpr.json

* Add condition for uwp patch

* Update versions/c-/cpr.json

* Update ports/cpr/portfile.cmake

* Update versions/c-/cpr.json

maxGimeno committed:
[ CGAL] Update version to 5.2.2 (#18387)

* Update to CGAL 5.2.2

* x-add-version

* Reset port version to 0

* x-add-version

* Update versions/c-/cgal.json

Co-authored-by: Phoebe <20694052+PhoebeHui@users.noreply.github.com>

Co-authored-by: Phoebe <20694052+PhoebeHui@users.noreply.github.com>

mathisloge committed:
[nanopb] update to 0.4.5 and add generator (#18241)

* [nanopb] update to 0.4.5 and add generator

* add versions

* fix python install?

* update version

* fix python? again

* version

* test python

* add version

* fix portfile

* version

* Update ports/nanopb/portfile.cmake

Co-authored-by: nicole mazzuca <83086508+strega-nil-ms@users.noreply.github.com>

* Update ports/nanopb/portfile.cmake

Co-authored-by: nicole mazzuca <83086508+strega-nil-ms@users.noreply.github.com>

* update version

Co-authored-by: nicole mazzuca <83086508+strega-nil-ms@users.noreply.github.com>

BillyONeal committed:
Update VMs and pick up VS2019 16.10 (#18233)

* Cherry-pick https://github.com/microsoft/vcpkg/pull/15598

* Hook deploy-inteloneapi into create-vmss.ps1.

* Add script to resolve https://github.com/microsoft/vcpkg/issues/17521

* Move tls settings deployment to the front and respond to script triggering a reboot.

* Go back to provisioning an extra disk to workaround https://github.com/microsoft/vcpkg/issues/18379

* Disallow public access to blob storage and require TLS 1.2.

* Update Pools.

* Update tool to 2021-06-19

* [simage] Skip simage on uwp platforms as it appears broken by 16.10.

* [tensorflow-cc] Skip because changes in our MacOS hardware broke the port.

autoantwort committed:
[vcpkg.cmake] PREPEND CMAKE_FIND_ROOT_PATH (#17336)

* [vcpkg.cmake] PREPEND CMAKE_FIND_ROOT_PATH

* [vcpkg.cmake] make PREPEND optional (default off, option VCPKG_PREFER_VCPKG_LIBS)

* Apply suggestions from code review

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* Update scripts/buildsystems/vcpkg.cmake

Co-authored-by: nicole mazzuca <83086508+strega-nil-ms@users.noreply.github.com>
Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

BillyONeal committed:
[vcpkg-unstable] Update agent pool name

JonLiu1993 committed:
[boringssl] update to newer commit (#18604)

* [boringssl] update to newer commit

* update version

NancyLi1013 committed:
[vcpkg baseline][kf5* librsvg] Add DISABLE_PARALLEL_CONFIGURE (#18600)

* [kf5*] Add DISABLE_PARALLEL_CONFIGURE

* Update version files

* [librsvg] Add DISABLE_PARALLEL_CONFIGURE

* Update version files

Neumann-A committed:
[benchmark] Update to 1.5.5 (#18595)

* update benchmark

* version stuff

dimztimz committed:
[nuspell] Update to version 5.0.0 (#18590)

* [nuspell] Update to version 5.0.0

* x-add-version nuspell

JackBoosY committed:
[matplotplusplus] Fix usage (#18583)

* [matplotplusplus] Fix usage

* update version record

* Remove rename code

* update version

JackBoosY committed:
[vcpkg baseline][ncurses] Fix gnu download link (#18579)

* [ncurses] Fix gnu download link

* update version record

Neargye committed:
[magic-enum] Update to v0.7.3 (#18575)

* [magic-enum] Update to v0.7.3

* fix

timmeh87 committed:
[libsmb2] New port (#18471)

* added libsmb2 port

* added version file

* updated baseline

* Ran command
./vcpkg x-add-version --all
 and then
./vcpkg format-manifest ports/libsmb2/vcpkg.json

* updated SHA based on result of "./vcpkg x-add-version libsmb2" and also added osx=fail to CI file in order to pass CI

* suggested changes made

* sha

* Update scripts/ci.baseline.txt

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

* Update ports/libsmb2/vcpkg.json

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

* Update ports/libsmb2/portfile.cmake

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

* sha

* added missing "vcpkg_check_linkage(ONLY_STATIC_LIBRARY)"

* sha

* vcpkg_fixup_pkgconfig()

* sha

Co-authored-by: Tim <tim@fake.WHAT>
Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

JonLiu1993 committed:
[fluidsynth] update to 2.2.1 (#18524)

* [fluidsynth] update to 2.2.1

* update version

* delete vcpkg_fixup_pkgconfig

* update version

an-tao committed:
[trantor] Update to 1.5.0 (#18534)

* [trantor] Update to 1.5.0

* Add vcpkg_fixup_pkgconfig()

evpobr committed:
[libogg] Upgrade to 1.3.5 (#18546)

* [libogg] Upgrade to 1.3.5

* Upgrade to 1.3.5.
* Convert CONTROL to vcpkg.json
* Change homepage to https://www.xiph.org/ogg (as in README.md)
* Download from the main repo https://gitlab.xiph.org/xiph/ogg

* [libogg] Use version instead of version-string in manifest

* [libogg] Update libogg.json

Sibras committed:
[fdk-aac] Update to version 2.0.2. (#18549)

* [fdkaac] Update to version 2.0.2.

* [fdk-aac] Remove unused files.

* Update fdk-aac.json

* Update ports/fdk-aac/portfile.cmake

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>

* test uwp|arm compilation.

* Update fdk-aac.json

* Convert to json.

* Update vcpkg.json

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>

c72578 committed:
[pcre] Update to 8.45 (#18550)

- New upstream version 8.45
  This is the final release of PCRE1
- Update patches:
  export-cmake-targets.patch
  pcre-8.45_suppress_cmake_and_compiler_warnings-errors.patch
- Add patch:
  pcre-8.45_fix_postfix_for_debug_Windows_builds.patch
- Remove deprecated SYSTEM_LIBRARIES from vcpkg_fixup_pkgconfig()
- Switch from CONTROL to vcpkg.json
    .\vcpkg format-manifest .\ports\pcre\CONTROL

luncliff committed:
[gstreamer] create a new port (#17394)

* [gstreamer] create port with meson

* initial build config
* nodownload in vcpkg_configure_meson needs be removed

* [gstreamer] remove duplicated GL headers

* 1st check for deps in windows
* mac: fix regex for bison

* [gstreamer] more detailed options

* remove glib usage

Will use WIP changes in microsoft/vcpkg

* [gstreamer] build with glib port

* [gstreamer] extract plugins to features

* fix some config errors in windows
* fails to find z.lib. this should be zlib.lib

* [gstreamer] create patches to search 'zlib'

* update tool list

* [vcpkg meson] add path of GIT

* use append for LINKER_FLAGS

Allow injection of LINKER_FLAGS to vcpkg_configure_meson

* [gstreamer] update baseline and port SHA

* [gstreamer] zlib is required for Windows

* Revert "[vcpkg meson] add path of GIT"

This reverts commit 8a67fbd1d287fa769cf516b834981ea175463492.

* [gstreamer] make git visible in meson

* [gstreamer] fix platform expression

* [gstreamer] revert change of vcpkg_configure_meson

* [gstreamer] modify build.ninja after configure

* [gstreamer] drop unknown flag with vcpkg_replace_string

* [gstreamer] disable 'devtools' option

* [gstreamer] remove dylib

* [gstreamer] move dll,pdb after install

* [gstreamer] remove 'nls' feature

Co-authored-by: Billy Robert ONeal III <bion@microsoft.com>

ankurvdev committed:
[python3] Fix python3:x64-linux builds (#18405)

* Fix build

* Update vcpkg.json

* Fix build

Co-authored-by: Ankur Verma <ankurv@microsoft.com>
Co-authored-by: Ankur Verma <ankurv@nuc.hogwarts>

BillyONeal committed:
CI Fixes from 2021-06-26 (#18664)

* CI Fixes from 2021-06-26

https://dev.azure.com/vcpkg/public/_build/results?buildId=55364&view=logs&j=0a61404f-5c45-5632-e83c-408b7fcca1d6&t=59cc5a87-398a-597b-cb3d-5842723775f1

PASSING, REMOVE FROM FAIL LIST: aubio:arm64-windows (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: aubio:x64-windows (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: aubio:x86-windows (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: caf:arm64-windows (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: chartdir:x64-linux (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: coolprop:x64-linux (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: fdk-aac:arm-uwp (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: fdk-aac:arm64-windows (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: fdk-aac:x64-uwp (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: fontconfig:x64-windows-static-md (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: google-cloud-cpp:arm64-windows (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: gtkmm:x64-linux (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: libmagic:x64-windows-static (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: libmagic:x64-windows-static-md (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: libopusenc:x64-linux (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: lua:arm-uwp (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: lua:x64-uwp (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: matio:x64-linux (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: offscale-libetcd-cpp:arm64-windows (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: proj4:arm64-windows (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: z3:arm64-windows (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: zeromq:arm64-windows (.\scripts\ci.baseline.txt)

:)

REGRESSION: clipp:x86-windows. If expected, add clipp:x86-windows=fail to .\scripts\ci.baseline.txt.

>[DEBUG] Downloading https://vcpkgteststorage.blob.core.windows.net/osxpr20201017/4645fafe85a8099ea97b85e939747a12e9b3b92213b5b8207a9c277537377b77b5daebd88a4c090ea89cfff2937a9fc155da6e8b5558574d7129227c28826e1c?sv=2019-12-12&ss=b&srt=sco&sp=rwlac&se=2021-06-01T19:00:00Z&st=2021-02-02T21:14:48Z&spr=https&sig=5OqlHLAKxxH9%2F2G%2BwO4ngXiG%2Bd0ASB81PklNDOIAUUk%3D
>[DEBUG] Downloading https://github.com/muellan/clipp/archive/2c32b2f1f7cc530b1ec1f62c92f698643bb368db.tar.gz
>Error: Failed to download from mirror set:
>https://vcpkgteststorage.blob.core.windows.net/osxpr20201017/4645fafe85a8099ea97b85e939747a12e9b3b92213b5b8207a9c277537377b77b5daebd88a4c090ea89cfff2937a9fc155da6e8b5558574d7129227c28826e1c?sv=2019-12-12&ss=b&srt=sco&sp=rwlac&se=2021-06-01T19:00:00Z&st=2021-02-02T21:14:48Z&spr=https&sig=5OqlHLAKxxH9%2F2G%2BwO4ngXiG%2Bd0ASB81PklNDOIAUUk%3D: failed: status code 403
>https://github.com/muellan/clipp/archive/2c32b2f1f7cc530b1ec1f62c92f698643bb368db.tar.gz: failed: status code 502
>[DEBUG] C:\A\1\14\s\src\vcpkg\base\downloads.cpp(652)

Looks like a transient download failure. Might be something funky with artifact caching.

REGRESSION: liblas:arm64-windows. If expected, add liblas:arm64-windows=fail to .\scripts\ci.baseline.txt.

>fatal error C1189: #error:  The file liblas/detail/endian.hpp needs to be set up for your CPU type.

Probably was blocked by one of the many "passing" libs on arm64-windows. Skipped.

REGRESSION: libsbml:x64-windows-static. If expected, add libsbml:x64-windows-static=fail to .\scripts\ci.baseline.txt.

>[DEBUG] Downloading https://vcpkgteststorage.blob.core.windows.net/osxpr20201017/49dedaa2fcd2077e7389a8f940adf931d80aa7a8f9d57330328372d2ac8ebcaeb03a20524df2fe0f1c6933587904613754585076c46e6cb5d6f7a001f427185b?sv=2019-12-12&ss=b&srt=sco&sp=rwlac&se=2021-06-01T19:00:00Z&st=2021-02-02T21:14:48Z&spr=https&sig=5OqlHLAKxxH9%2F2G%2BwO4ngXiG%2Bd0ASB81PklNDOIAUUk%3D
>[DEBUG] Downloading https://sourceforge.net/projects/sbml/files/libsbml/5.18.0/stable/libSBML-5.18.0-core-plus-packages-src.tar.gz/download
>Error: Failed to download from mirror set:
>https://vcpkgteststorage.blob.core.windows.net/osxpr20201017/49dedaa2fcd2077e7389a8f940adf931d80aa7a8f9d57330328372d2ac8ebcaeb03a20524df2fe0f1c6933587904613754585076c46e6cb5d6f7a001f427185b?sv=2019-12-12&ss=b&srt=sco&sp=rwlac&se=2021-06-01T19:00:00Z&st=2021-02-02T21:14:48Z&spr=https&sig=5OqlHLAKxxH9%2F2G%2BwO4ngXiG%2Bd0ASB81PklNDOIAUUk%3D: failed: status code 403
>https://sourceforge.net/projects/sbml/files/libsbml/5.18.0/stable/libSBML-5.18.0-core-plus-packages-src.tar.gz/download: WinHttpReceiveResponse() failed: 12175
>
>[DEBUG] C:\A\1\14\s\src\vcpkg\base\downloads.cpp(652)
>[DEBUG] Exiting after 548953 us (542540 us)

Looks like a transient download failure. But 403 forbidden looks like artifact caching failure...

REGRESSION: luafilesystem:arm-uwp. If expected, add luafilesystem:arm-uwp=fail to .\scripts\ci.baseline.txt.
REGRESSION: luafilesystem:x64-uwp. If expected, add luafilesystem:x64-uwp=fail to .\scripts\ci.baseline.txt.
REGRESSION: luasocket:arm-uwp. If expected, add luasocket:arm-uwp=fail to .\scripts\ci.baseline.txt.
REGRESSION: luasocket:x64-uwp. If expected, add luasocket:x64-uwp=fail to .\scripts\ci.baseline.txt.

Were previously blocked by lua, marked as 'fail'.

REGRESSION: vcpkg-find-acquire-program:arm-uwp. If expected, add vcpkg-find-acquire-program:arm-uwp=fail to .\scripts\ci.baseline.txt.
REGRESSION: vcpkg-find-acquire-program:x64-windows-static. If expected, add vcpkg-find-acquire-program:x64-windows-static=fail to .\scripts\ci.baseline.txt.
REGRESSION: vcpkg-find-acquire-program:x64-windows. If expected, add vcpkg-find-acquire-program:x64-windows=fail to .\scripts\ci.baseline.txt.

These are almost certainly the same artifact caching transient.

* Aubio typo.

dg0yt committed:
[netcdf-c] Fix pc files; use features (#18120)

* Format manifest

* New port-version

* Quote filepath expressions

* Port to vcpkg-cmake ports

* Remove invalid nc-config file

* Fix pc files

* Use features

* Add feature 'tools'

* x-add-version

* Fix build with msys2 mingw

* Update git-tree

h3ndrk committed:
[portaudio] Switch to vcpkg_from_github, minor format improvements (#18065)

* Add portaudio:: namespace

* Switch from vcpkg_from_git to vcpkg_from_github

* Convert portfile to unix line-endings

* Increase port-version of portaudio

* Update version hash of portaudio

* Remove namespace patch

Signed-off-by: Hendrik <git@h3ndrk.de>

* Update version hash of portaudio

* Fix indent issue

* Update the baseline version

* Fix the port version

* Fix baseline version

* Remove redundant tag in portfile

* Update version hash of portaudio

Co-authored-by: PhoebeHui <20694052+PhoebeHui@users.noreply.github.com>

luncliff committed:
[sleef] create a new port (#18075)

* [sleef] create a new port

* [sleef] update baseline

* [sleef] update supports

* add more supported architectures to sleef

* [sleef] update ci.baseline.txt

* Revert "[sleef] update ci.baseline.txt"

This reverts commit 39685ae755048ebc4702ddfb659026f910ce885a.

* [sleef] disable (arm & windows)

The project's CI doesn't check it. Disable in support field.

* Update ports/sleef/vcpkg.json

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

* [sleef] update git-tree SHA

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>
Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

BillyONeal committed:
Update vcpkg tool to resolve leak of the asset cache token (#18720)

yurybura committed:
[llvm] update to 12.0.0, enable zlib and libxml2 (#17302)

* [llvm] update to 12.0.0, enable zlib and libxml2

* [llvm] disable libxml2 by default, arm64-windows should be supported

* [llvm] try to fix build with ffi on Windows #17663

* [llvm] re-enable libxml2

* [llvm] fix formatting

* [llvm] overwrite version

* [mesa] update to v21.1.0

* [llvm] update ci.baseline

* [halide] update to the latest master commit

* [llvm] disable libxml2 port until complete PR #15390

* update versions

* [llvm] fix disabling of external libraries

* update version

* update ci.baseline

* update ci.baseline

* fix typo

* [llvm] fix libxml2 (depends on #17945)

* revert "[halide] update to the latest master commit"

* update versions

* [halide] fix zlib dependency for tools

* update versions

* [halide] update to official release v12.0.0

* update versions

* [halide] update to v12.0.1

* update version

* [mesa] remove llvm from defaults

* update version

* update ci.baseline

* [mesa] update to v21.1.2 and fix build with LLVM on Windows

* [llvm] allow to build libunwind project on Windows

* update versions

* [mesa] update ci.baseline.txt to disable x64-windows-static-md, may be fixed in the PR #18495

BillyONeal committed:
[vcpkg internal] Update compiler test scripts to use asset caching.

evpobr committed:
[mpg123] Upgrade to 1.28.0 (#18403)

* [mpg123] Upgrade to 1.28.0

* [libopenmpt] Upgrade to 2017-01-28-cf2390140-2

* [mpg123] Disable UWP port

* Fix libmpg123 UWP build

* Fix libout123 UWP build

* Fix libsyn123 UWP build

* Update ports/libopenmpt/CMakeLists.txt

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

* Update ports/libopenmpt/CMakeLists.txt

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

* [libopenmpt] Update libopenmpt.json

* [mpg123] Separate UWP patches

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>
Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

dg0yt committed:
[vcpkg_configure_make] Fix mingw (cross) build issues (#18554)

* Set ENV{WINDRES} for mingw

* Fix passing of '${prefix}' for mingw cross builds

ysc3839 committed:
[vcpkg] Add static triplet for arm-windows (#18300)

autoantwort committed:
[scripts] Add vcpkg_completions.zsh (#18274)

* Add vcpkg_completions.zsh

* Update scripts/vcpkg_completion.zsh

NancyLi1013 committed:
[vcpkg baseline][simage] Update to 1.8.1 (#18645)

* [simage] Update to 1.8.1

* Format manifest and update version files

* Revert the changes for arm and uwp in ci.baseline.txt

* Update versions/s-/simage.json

* Update ports/simage/portfile.cmake

* Update ports/simage/portfile.cmake

* Update ports/simage/portfile.cmake

* Update versions/s-/simage.json

* Add options to packages

* Update versions/s-/simage.json

* Update portfile.cmake

* Update versions/s-/simage.json

* Update portfile.cmake

* Update versions/s-/simage.json

* [simage] Reorganize all features, add features

* Install tool simage-config on non-Windows

* fix typo

* Add supports field and update ci.baseline.txt

* Update versions/s-/simage.json

Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>
Co-authored-by: JackBoosY <yuzaiyang@beyondsoft.com>

dg0yt committed:
[curl] Fix linker flags in curl-config and libcurl.pc for osx (#17790)

* Fix exported framework linker flags for osx

* x-add-version

* Remove cmake version check

* Update git-tree

alexreinking committed:
[halide] Fix several issues with the Halide port after v12 (#18747)

* Fix several issues with the Halide port after v12

1. Use vcpkg-cmake and vcpkg-cmake-config ports
2. Fix usage file to point to relevant documentation
3. Remove OpenGL feature (Halide 12 dropped support)
4. Drop VCPKG_POLICY_EMPTY_PACKAGE
5. Set new Halide packaging variables for better vcpkg compliance.

* update versions

NancyLi1013 committed:
[nettle] Remove from ci.baseline.txt (#18745)

* [nettle] Test on x64 windows

* Revert the changes for nettle

aggieNick02 committed:
[tbb] Remove ability to disable exception support in TBB runtime (#18740)

* After filing an issue with TBB (https://github.com/oneapi-src/oneTBB/issues/414), I found out that building the library with exceptions explicitly disabled as I had added support for had gone from not officially supported to more explicitly disallowed

* update tbb port-version

* run x-add-version

kayru committed:
[libressl] update to 3.3.3 (#18729)

Resolves #18682

mathisloge committed:
[asio] update to 1.18.2 (#18728)

* [asio] update to 1.18.2

* update versions

eao197 committed:
[json-dto] Update to 0.2.13 (#18710)

* json_dto updated to 0.2.13.

* x-add-version for json-dto.

goyarn committed:
[README_ko_KR.md] Fix typo for korean doc (#18673)

원도우 -> 윈도우

cffk committed:
[geographiclib] Update to version 1.52 (#18694)

* [geographiclib] Update to version 1.52

* run ./vcpkg x-add-version geographiclib

* ports/geographiclib/vcpkg.json: version-string -> version

* Another iteration with x-add-version

dkavolis committed:
[spdlog] Do not build examples (#18639)

* Don't build spdlog examples

* Update port version

an-tao committed:
[drogon] Update to 1.7.1 (#18643)

JonLiu1993 committed:
[socket-io-client] update to 3.0.0  (#18646)

* [socket-io-client] update to 3.0.0

* update version

* update patch file

* update version

kevinhartman committed:
[morton-nd] Update to v4.0.0. (#18652)

* Update morton-nd to v4.0.0.

* Update baselines.

NancyLi1013 committed:
[libdatachannel] Fix usage (#18621)

* [libdatachannel] Fix usage

* Update version files

PazerOP committed:
[fmt] Fix an easily avoidable compilation warning (#18622)

* Fix an unconditional #pragma warning breaking other compilers.

* Increase fmt version

* re-ran vcpkg x-add-version --all --overwrite-version, this time it produced a different result for some reason

AenBleidd committed:
[sentry-native] Update to 0.4.10 (#18627)

Signed-off-by: Vitalii Koshura <lestat.de.lionkur@gmail.com>

bernhardmgruber committed:
[Vc] Update to 1.4.2 (#18601)

Commit:
[seal] Fixed cannot find SEAL. (#18596)

* Fixed cannot find seal.

* Updated versioning.

ahojnnes committed:
[mongo-c-driver] Fix for custom VCPKG_BUILD_TYPE (#18569)

* Fix mongo-c-driver for custom VCPKG_BUILD_TYPE

* Update version

* Update ports/mongo-c-driver/portfile.cmake

* Update versions/m-/mongo-c-driver.json

Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>
Co-authored-by: Billy Robert ONeal III <bion@microsoft.com>

bobuc-havok committed:
[protobuf] Fixed protoc path (#18741)

* fixed protoc path

* bumped version

JackBoosY committed:
[recast] Export unofficial target (#18671)

* [recast] Export unofficial target

* update version record

* Export INTERFACE_INCLUDE_DIRECTORIES

* Export more include dirs.

* Update versions/r-/recast.json

sandercox committed:
[vcpkg] Expand Architecture list with escape chars (#18156)

* Expand Architecture list with escape chars

This expands the architecture lists with escape characters. When
building FAT binaries for macos using multiple architectures in the
values they need to be escaped otherwise they are passed on to CMake
incorrectly #14932

* Adding the architecture fix to vcpkg-cmake port
updating port vcpkg-cmake version

* updated version in baseline

omartijn committed:
[ffmpeg] multi arch support (#18635)

* Expand Architecture list with escape chars

This expands the architecture lists with escape characters. When
building FAT binaries for macos using multiple architectures in the
values they need to be escaped otherwise they are passed on to CMake
incorrectly #14932

* Adding the architecture fix to vcpkg-cmake port
updating port vcpkg-cmake version

* updated version in baseline

* Allow building ffmpeg for multiple architectures on macOS

Since ffmpeg does _not_ support multi-arch builds due to their use of
autotools and config.h which then includes the wrong platform-dependent
functions, we must perform two separate builds and join them using lipo

* fixup! Allow building ffmpeg for multiple architectures on macOS

* fixup! Allow building ffmpeg for multiple architectures on macOS

Co-authored-by: Sander Cox <sander@paralleldimension.nl>
Co-authored-by: Martijn Otto <martijn@resolume.com>

bobuc-havok committed:
[libxml2] Fix target (#18714)

* fix LibXml2 target dependencies

* updated version files

NancyLi1013 committed:
[bzip2] Update repository (#18705)

* [bzip2] Update repository

* Update version files

JackBoosY committed:
[tgui] Fix feature tool dependency (#18724)

* [tgui] Fix feature tool dependency

* update version record

lstipakov committed:
[OpenSSL] support for "no-autoload-config" config option (#18389)

* [OpenSSL]: support for "no-autoload-config" config option

OpenSSL has "no-autoload-config" config option,
which disables automatic config loading:

  https://github.com/openssl/openssl/pull/5959

This patch allows to set above option via variable
in custom triplet:

  # build-openssl-no-autoload-config.cmake
  set(OPENSSL_NO_AUTOLOAD_CONFIG ON)

and use it like:

  vcpkg install --overlay-triplets=<path-to-triplets> \
    openssl:build-openssl-no-autoload-config

Signed-off-by: Lev Stipakov <lev@openvpn.net>

* [OpenSSL] bump version

Signed-off-by: Lev Stipakov <lev@openvpn.net>

mikke89 committed:
[rmlui] Add Rmlui library (#18409)

* Add RmlUi port

* Update versions

* Fix uwp build

* Update versions

* Add maintainers to manifest

* Update version

* Update ports/rmlui/portfile.cmake

Co-authored-by: Noam Chitayat <noam-c@users.noreply.github.com>

* - Update to RmlUi 4.1, fixes some issues with missing dependencies.
- Add robin-hood-hashing as dependency, replacing built-in copy.
- Only need FreeType core.
- Static library definition now replaced in-source.

* Fix config name

* Change source in current packages directory

* Use robin_hood.h included from vcpkg

* Move source modifications to robin_hood patch

* Fix path in patch

Co-authored-by: Noam Chitayat <noam-c@users.noreply.github.com>

mathijs727 committed:
[cnl] Update to version 1.1.7 (#18439)

* [cnl] update to version 1.1.7

* Update baseline.json

* Update ports/cnl/CONTROL

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

* Update ports/cnl/portfile.cmake

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

* Update HEAD_REF to point to main

* Bump port-version

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

* Run vcpkg x-add-version cnl

* Update versions/c-/cnl.json

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

* Update ports/cnl/CONTROL

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

* Update versions/baseline.json

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

* Update versions/c-/cnl.json

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

* Update git tree with vcpkg x-add-version

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>
Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

strega-nil committed:
[vcpkg.cmake] fix X_VCPKG_APPLOCAL_DEPS_INSTALL (#18607)

* [vcpkg.cmake] fix X_VCPKG_APPLOCAL_DEPS_INSTALL

* Billy CR

JonLiu1993 commented:
@mheyman ，Are you still working on fixing this pr?

JPMMaia committed:
[klein] create a new port  (#18279)

* [klein] new port

* [klein] Replace version-string by version

* [klein] Add call to vcpkg_fixup_pkgconfig()

* [klein] Replace klein by ${PORT}

* [klein] Add patch to avoid using FetchContent

* [klein] Change version to version-date

* update version

* Apply suggestions from code review

Co-authored-by: nicole mazzuca <83086508+strega-nil-ms@users.noreply.github.com>

* Update git-tree version

Co-authored-by: Jonliu1993 <13720414433@163.com>
Co-authored-by: nicole mazzuca <83086508+strega-nil-ms@users.noreply.github.com>

mheyman commented:
Been away - still planning on it. I started the 3-port solution but the
patches required were not quick to finish and I need to set up some tests
too.

On Tue, Jul 6, 2021 at 2:55 AM JonLiu1993 ***@***.***> wrote:

> @mheyman <https://github.com/mheyman> ，Are you still working on fixing
> this pr?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/microsoft/vcpkg/pull/18411#issuecomment-874508518>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJT7J2A3CIS755PLEOHHPLTWKSGBANCNFSM46S5VMSA>
> .
>

PhoebeHui committed:
[vcpkg baseline][cnl] Disable tests (#18767)

dg0yt committed:
[hdf5] Fix linking order in pc file (#18825)

* Fix link order in pc file

* x-add-version

bansan85 committed:
[catch2] Fix linux build in NTFS fs (#18814)

file(RENAME Camel lower) works fine when running vcpkg in
Windows Powershell with triplet x64-windows in NTFS fs.

file(RENAME Camel lower) does not work fine when running vcpkg in
WSL with triplet x64-linux in NTFS.

julianxhokaxhiu committed:
[binarycaching.md] Improve Azure DevOps example (#18807)

Add a missing, but required task for Azure DevOps Pipelines, to automatically authenticate against the Artifacts registry.
This should help any vcpkg newcomer to integrate easily.

MCJack123 committed:
[sdl2-mixer] Add fluidsynth feature  (#18826)

* [sdl2-mixer] Add fluidsynth feature

* [sdl2-mixer] Add fluidsynth feature

* [sdl2-mixer] Add fluidsynth feature

* Updated versions & fixed style

* Fixed version hash

JonLiu1993 committed:
[paho-mqtt] update to 1.3.9 (#18765)

* [paho-mqtt] update to 1.3.9

* update version

* revert patch format

* update version

Neumann-A committed:
[Qt6] Update to 6.1.2 (#18787)

* update qt to 6.1.2

* version stuff

mcmtroffaes committed:
[ffmpeg] fix LGPL build, disable avisynthplus on static builds (#18803)

* [ffmpeg] do not pull in postproc for LGPL build

* [ffmpeg] disable avisynthplus on static builds (as it is not functional)

* [ffmpeg] bump port version

* [ffmpeg] x-add-version

coryan committed:
[google-cloud-cpp] update to latest release (v1.29.0) (#18759)

JonLiu1993 committed:
[exiv2] update to 0.27.4 (#18750)

* [exiv2] update to 0.27.4

* update version

luca-schlecker committed:
[crow] update to v0.3+1 (#18706)

Signed-off-by: Luca Schlecker <luca.schlecker@hotmail.com>

shybovycha committed:
[irrlicht] Add OSX support (#18725)

* Fix Irrlicht compilation on OSX

* Update port description

* Update portfile instructions

* Update version

* Fix patch files

* Add missing cmake dependencies

* Remove obsolete comment

* Update version

* Remove OBJCXX language from project definition to prevent Windows builds from failing

* Update versions

* Code review tweaks

* Update version

* Update versions/i-/irrlicht.json

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

* Fix qt5-3d build failure on OSX

* Update qt5-3d version

* Update port-version for qt5-3d

* Update version for qt5-3d

* [qt5-3d] Revert modification to previous version entry

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>
Co-authored-by: Robert Schumacher <ras0219@outlook.com>

mathisloge committed:
[mapbox-wagyu] add new port (#18771)

* add wagyu

* add versions

* update hash

* update versions

myd7349 committed:
[ftxui] Add new port (fix #16327) (#18360)

* [ftxui] Add new port (fix #16327)

* [ftxui] x-add-version ftxui

* [ftxui] Drop UWP support

* [ftxui] Update to 2021-06-16

* [ftxui] Overwrite version

* [ftxui] Try to fix linux and macOS build

* [ftxui] Overwrite version

* [ftxui] Fail fast on UWP

* [ftxui] Clean version file

* [ftxui] Overwrite version

* [ftxui] Update to 2021-06-17

* [ftxui] Overwrite version

* [ftxui] Remove debug/share

* [ftxui] Overwrite version

* [ftxui] Quote paths

* [ftxui] Overwrite version

playgithub committed:
[xlsxio] new port (#18438)

* add port: xlsxio

* vcpkg x-add-version xlsxio

* fix deps

* add vcpkg_fixup_pkgconfig()

* Apply suggestions from code review

To be consistent, `OUT_FEATURE_OPTIONS FEATURE_OPTIONS` in a new line, as in `vcpkg_from_git`.

Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>

* Update ports/xlsxio/portfile.cmake

Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>

* format

* use date as version string

* vcpkg format-manifest

* Update ports/xlsxio/portfile.cmake

Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>

* patch for dependency

* fix typo

* vcpkg x-add-version

* remove feature wide

* Update ports/xlsxio/vcpkg.json

deprecate libzip that's an alternative and imperfect

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

* Update ports/xlsxio/vcpkg.json

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

* Update versions/x-/xlsxio.json

Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>

* Remove x64-windows-cpp17 community triplet

Co-authored-by: abc <a@b.com>
Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>
Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

omartijn committed:
[boost-modular-build-helper] Fix boost build toolchain options not being used (#18529)

* Fix boost build toolchain options not being used

The issue was due to the options only being set for the gcc toolchain,
and then only for a specific version. On platforms defaulting to a
different toolchain (e.g. macOS) this didn't work at all.

Additionally, some missing flags were not propagated, in particular the
CMAKE_OSX_DEPLOYMENT_TARGET, CMAKE_OSX_SYSROOT and CMAKE_OSX_ARCHITECTURES

* [boost-modular-build-helper] Apply CR comments

Co-authored-by: Martijn Otto <git@martijnotto.nl>
Co-authored-by: Robert Schumacher <ras0219@outlook.com>

ctabin committed:
[libzippp] Update to 4.1-1.8.0 (#18553)

Co-authored-by: Cédric Tabin <cedric.tabin@gmail.com>

shishirb-MSFT committed:
[proxywrapper] Remove proxywrapper package (#18692)

* Remove proxywrapper package

* [proxywrapper] Remove from baseline since there is no longer a tested version of the port

Co-authored-by: Robert Schumacher <ras0219@outlook.com>

PhoebeHui committed:
[glew] Improve messages for system opengl dependency (#18672)

* [glew] Fix the dependency

* Update baseline version

* Adress the review suggestions

* Update the baseline revision

JonLiu1993 committed:
[socket-io-client]fix socket-io-client use  (#18768)

* update patch file

* update version

autoantwort committed:
[boost] update to 1.76.0 (#17335)

* [boost] update generator script for boost 1.76

* [boost] update ports to 1.76.0 (run generator)

* [boost] fix windows build?

* [quantlib] update and fix mac build

* [symengine] update and fix build

* [avro-cpp] update to latest master and fix windows build

* [folly] update to 2021.05.31.00

* [fbthrift, fizz, wangle] update to v2021.05.31.00 and fix build

* [proxygen] update to version 2021.05.31.00

* [fizz, proxygen, fbthrift] fix sodium target

* [proxygen] also works on macOS

* [quantlib] use fix from upstream to fix mac build

* [symengine] minimize patch file and fix deprecation warning

* [folly,proxygen,wangle,fizz,fbthrift] update to 2021.06.14.00

* [fbthrift] remove unnecessary dependency rsocket

I couldn't find any information that this dependency exists. The term is used in the code, but not in the context of a dependency

* [fizz,fbthrift] fix zlib dependency

* [fbthrift] pass required flex executable to cmake configure

* add version files

* [boost] generate-ports.ps1: Apply code review

* [boost] changes from new version of generate-ports script

* update version files

* [boost] generate-ports.ps1: Apply code review

simogasp committed:
[cctag] Add new port (#16719)

* [cctag] add port

* [cctag] using version-semver

* [cctag] fix for multithreaded compilation

* [cctag] update ci.baseline

* [cctag] fix format json

* [cctag] fix version with semver

* [cctag] update version sha

* [cctag] v1.0.0-rc2

* [cctag] v1.0.0-rc3 with vcpkg_copy_tools(), much simpler

* [cctag] update port file

* [cctag] v1.0.0-rc4

* [cctag] update version file

* [cctag] v1.0.0

* [cctag] update version file v1.0.0

yourWaifu committed:
[sleepy-discord] Add new port (#17604)

* New port sleepy-discord

* Add versions for Sleepy Discord

* SleepyDiscord: use website as homepage

* SleepyDiscord: Remove No newline at end git diff

* SleepyDiscord: format manifest

* SleepyDiscord: rerun x-add-version

* SleepyDiscord: add compression feature

* SleepyDiscord: Update version string

* SleepyDiscord: rerun x-add-version again

* SleepyDiscord: set version-date

* SleepyDiscord: rerun add-version

* SleepyDiscord: Add requested changes

* SleepyDiscord: run add version

* SleepyDiscord: update library

* SleepyDiscord: rerun x-add-version

* SleepyDiscord: copy config file to share folder

* SleepyDiscord: run add-version

* SleepyDiscord: Fix issue with parallel configuring

* SleepyDiscord: x-add-version

* SleepyDiscord: replace deprecated functions

* SleepyDiscord: x-add-version

* SleepyDiscord: Install config during configure

* SleepyDiscord: x-add-version

* [sleepy-discord] update: add Select Menus

* [sleepy-discord] run add-version

* [sleepy-discord] rerun add-verion

BillyONeal committed:
Download khrplatform.h with vcpkg instead of gl3w_gen.py so that our downloader settings are used (#18858)

Alternative resolution of https://github.com/microsoft/vcpkg/pull/18846/

azure-sdk committed:
[azure-core-cpp] Update to 1.1.0 (#18779)

* [azure-core-cpp] Update to 1.1.0
## 1.1.0 (2021-07-02)

### Bugs Fixed

- Fixed a memory leak issue in `Base64Encode()`. (A community contribution, courtesy of _[jorgen](https://github.com/jorgen)_)

### Other Changes

- Made internal-only changes to support the Azure Key Vault client library.

### Acknowledgments

Thank you to our developer community members who helped to make Azure Core better with their contributions to this release:

- Jorgen Lind _([GitHub](https://github.com/jorgen))_

* [azure-storage-common-cpp] Update to 12.0.1
## 12.0.1 (2021-07-07)

### Bug Fixes

- Fixed a memory leak issue while parsing XML.

* [azure-storage-blobs-cpp] Update to 12.0.1
## 12.0.1 (2021-07-07)

### Bug Fixes

- Fixed a bug where transactional MD5 hash was treated as blob MD5 hash when downloading partial blob.

* [azure-storage-files-datalake-cpp] Update to 12.0.1
## 12.0.1 (2021-07-07)

### Bug Fixes

- Fixed a bug where transactional MD5 hash was treated as blob MD5 hash when downloading partial blob.

* [azure-storage-files-shares-cpp] Update to 12.0.1
## 12.0.1 (2021-07-07)

No API changes since `12.0.0`.

* Remove curl version from vcpkg manifest

* Update version files

Co-authored-by: Anton Kolesnyk <antkmsft@users.noreply.github.com>

PragmaTwice committed:
[protopuf] Update to 2.1.0 (#18757)

* [protopuf] update to 2.1.0

* [protopuf] add version info

PhoebeHui committed:
[utf8proc] Export cmake targets (#18860)

* [utf8proc] Export cmake targets

* Update the baseline version

jixingcn committed:
[gli] upgrade the gli to latest (#18715)

* upgrade the gli to latest

* update the version of gli

* update the version of gli

* [gli] Revert patch, use date scheme

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

mathisloge committed:
[protozero] update to 1.7.0 (#18842)

* update protozero to 1.7.0

* update versions

* apply suggestions

* version

AenBleidd committed:
[sentry-native] Update to 0.4.11 (#18856)

Signed-off-by: Vitalii Koshura <lestat.de.lionkur@gmail.com>

Commit:
[ avcpp ] Update version (#18436)

* remove avcpp patch that was accepted upstream.

revert port version

* update versions

mheyman committed:
[daw-json-link daw-utf-range daw-header-libraries] new port

mheyman commented:
Renamed the port to `daw-json-link` which now depends on `daw-utf-range` which depends on `daw-header-libraries`.

`daw-utf-range` had been including a modified version of `utfcpp` - I moved the files under `include/daw` and the `utf8` namespace under `daw::` to keep from conflicting with the `utfccp` port. The modifications are constexpr/noexcept changes.

All of `daw-json-link` tests pass with the changes. Whew.

JonLiu1993 commented:
@mheyman ,Can you please resolve the conflicts against master?

JackBoosY commented:
Can you please merge to master?

mheyman committed:
Merge remote-tracking branch 'upstream/master'

mheyman commented:
Merged to master...

JackBoosY commented on versions/j-/json-link.json:
```diff
@@ -0,0 +1,9 @@
+{
```
Please remove this file.

JackBoosY commented on ports/daw-header-libraries/portfile.cmake:
```diff
@@ -0,0 +1,16 @@
+    HEAD_REF master
+)
+
+file(
+    COPY 
+        ${SOURCE_PATH}/include/daw
+    DESTINATION 
+        ${CURRENT_PACKAGES_DIR}/include/
+)
```
I found the upstream provides the CMakeLists.txt, can we use `vcpkg_cmake_configure` and `vcpkg_cmake_install` instead of this?

JackBoosY commented on ports/daw-json-link/portfile.cmake:
```diff
@@ -0,0 +1,41 @@
+    PATCHES moved_dragonbox_and_utfcpp.patch
+)
+
+file(
+    COPY 
+        ${SOURCE_PATH}/include/daw
+    DESTINATION 
+        ${CURRENT_PACKAGES_DIR}/include/
+)
+file(
+    COPY 
+        ${SOURCE_PATH}/include/third_party
+    DESTINATION 
+        ${CURRENT_PACKAGES_DIR}/include/daw
+)
```
See above.

JackBoosY commented on ports/daw-utf-range/portfile.cmake:
```diff
@@ -0,0 +1,19 @@
+    PATCHES my_utfcpp.patch
+)
+
+file(
+    COPY 
+        ${SOURCE_PATH}/include/utf8.h
+        ${SOURCE_PATH}/include/utf8
+        ${SOURCE_PATH}/include/daw/utf_range
+    DESTINATION 
+        ${CURRENT_PACKAGES_DIR}/include/daw
+)
```
See above.

JackBoosY commented on ports/daw-utf-range/my_utfcpp.patch:
```diff
@@ -0,0 +1,119 @@
+ #include <stdexcept>
+ 
+-namespace utf8 {
++namespace daw::utf8 {
```
I think we shouldn't modify the source code here.

mheyman commented:
The first route I went down was simply using CMakeLists.txt. When that failed in the github vcpkg build system I had to split the single port into the three, *daw-json-link*, *daw-utf-range*, and *daw-header-libraries*. And, I tried to go the route of patching the CMakeLists.txt in each of those along with modifying the sources as needed. After going through multiple debug cycles, I thought "hey, this is *header-only*, why am I putting myself through all this grief when I can simply do a `file(COPY...)`? So, I switched tack and in less than an hour, I had a working install.

If you really want me to go back to the CMakeList.txt solution, I think it wasn't too far from a working solution using patched versions of those. If you simply want a *daw-json-link-config.cmake* and the like, I can certainly through some of those in there but I generally don't use them for header-only libraries (header-only-library CMake config files are not too complicated to hand code).

JackBoosY commented:
@mheyman I will give you some help.

Commit:
[daw-header-libraries daw-utf-range daw-json-link]: vcpkg_cmake_configure, vcpkg_cmake_install, vcpkg_cmake_config_fixup

mheyman committed:
Merge remote-tracking branch 'upstream/master'

mheyman commented:
Now uses `vcpkg_cmake_configure()`, `vcpkg_cmake_install()`, `vcpkg_cmake_config_fixup()` instead of the simple header-file-copy solution. :-)

Not sure I like the highly-patched CMakeLists.txt solution over the header file copy. I understand it is idiomatic vcpkg to use the cmake helpers but the vcpkg port maintainability suffers - if Darrell Wright ever appreciably changes the CMakeLists.txt files in the source repositories, maintaining the patches becomes more difficult.

Commit:
[daw-json-link daw-utf-range daw-header-libraries] vcpkg x-add-version

JackBoosY commented on ports/daw-utf-range/utfcpp_and_portname.patch:
```diff
@@ -0,0 +1,187 @@
+ #include <ciso646>
+ #include <stdexcept>
+ 
+-namespace utf8 {
++namespace daw::utf8 {
```
I still don't know why you added an extra namespace `daw`?

mheyman replied:
The daw::utf8 namespace is there to keep from conflicting with the `utfcpp` vcpkg port. The source from that port is the same source that Darrell Wright modified with `constexpr`s and `noexcept`s (among other things) to work with his JSON library. I didn't want somebody to not be able to use both `utfcpp` *and* `json-link` in the same project. I considered just changing the namespace to `daw_utf8` but since I moved it under the `daw/` directory, I thought making namespace `daw::utf8` more appropriate.

JackBoosY replied:
Any changes related to the source code require upstream approval, can you report it upstream?

Thanks.

mheyman replied:
Done!

beached replied:
I have gone through the changes and updated upstream for daw-json-link, daw-header-libraries, and daw-utf-range.  I think that should reduce/remove the need to patch the source

Commit:
Remove useless version file

beached commented:
One comment on the testing.  It is primarily tested on x64 linux/macos/windows in c++17, in develop C++20 too.  It has less primarily tested on 32bit arm/32bit windows but not as a first class.  When travis had free CI it was tested on BE ppc and/or S390 I think.  It's tested in constant expressions so should work on all the triplets, but I don't know that with certainty

mheyman committed:
[daw-header-libraries daw-utf-8 daw-json-link] newest source

mheyman committed:
[daw-header-libraries daw-utf-range daw-json-link] vcpkg x-add-version

mheyman committed:
Merge remote-tracking branch 'upstream/master'

mheyman committed:
Merge branch 'master' of github.com:mheyman/vcpkg

mheyman commented:
The latest pull request includes the excellent source updates from [beached](https://github.com/beached) that removed any need for patch files.

mheyman committed:
Merge remote-tracking branch 'upstream/master'

JackBoosY commented on ports/daw-header-libraries/portfile.cmake:
```diff
@@ -0,0 +1,16 @@
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/${PORT})
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug" "${CURRENT_PACKAGES_DIR}/lib")
```
Why delete debug and lib?

mheyman replied:
At one point, that was required because the build created empty directories that vcpkg did not approve of. I admit I didn't recheck that after updating to the new daw code. I will recheck as soon as I can

JackBoosY replied:
If they are header-only library, please add comment at top of portfile.cmake.

Commit:
[daw-header-libraries daw-json-link daw-utf-range] doc: comment as header-only in portfile.cmake

mheyman committed:
Merge remote-tracking branch 'upstream/master'

Commit:
[daw-header-libraries daw-json-link daw-utf-range] --overwrite-version x-add-version

mheyman commented:
Header-only comments applied. Sorry for the delay. Life intervened but I did find a second flock of chickens in the neighborhood :-)

JackBoosY commented on versions/d-/daw-header-libraries.json:
```diff
@@ -0,0 +1,14 @@
+      "git-tree": "6687948f2f2f4c3ba578a8ac41624c58a541f1a2",
+      "version-semver": "1.21.1",
+      "port-version": 0
+    },
+    {
+      "git-tree": "ac2e26b64cc9c547c9d78fa19cf733d2c6b6a064",
+      "version-date": "2021-06-28",
+      "port-version": 0
+    }
```
```suggestion
    }
```

JackBoosY commented on versions/d-/daw-json-link.json:
```diff
@@ -0,0 +1,14 @@
+      "git-tree": "04d0df53fc3b69309aa10f53cc92f0933e8139b8",
+      "version-semver": "2.9.11",
+      "port-version": 0
+    },
+    {
+      "git-tree": "0285ca0ade967334ccf07c69e803c1435da01147",
+      "version-date": "2021-05-16",
+      "port-version": 0
+    }
```
```suggestion
    }
```

JackBoosY commented on versions/d-/daw-utf-range.json:
```diff
@@ -0,0 +1,14 @@
+      "git-tree": "e1c08fdf12013384e9a86e5233245321fea58793",
+      "version-semver": "2.1.0",
+      "port-version": 0
+    },
+    {
+      "git-tree": "c92a96715337b9019c1c12b2dae9596ec57c6dcc",
+      "version-date": "2021-06-15",
+      "port-version": 0
+    }
```
```suggestion
    }
```

JackBoosY committed:
Update versions/d-/daw-header-libraries.json

JackBoosY committed:
Update versions/d-/daw-json-link.json

JackBoosY committed:
Update versions/d-/daw-utf-range.json
